peer/peerdiscovery_test.go

```diff
@@ -0,0 +1,66 @@
+package peer_test
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/renproject/aw/wire"
+	"github.com/renproject/id"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Peer", func() {
+	Context("when trying to discover other peers using the peer discovery client", func() {
+		It("should successfully find all peers", func() {
```

Comment:
This is a great end-to-end test. However, it only checks that discovery works as expected when the nodes start in a ring topology. Consider adding a wrapping `Context("when peers begin in a ring topology")` and then add a few more topologies to this test. What about a star topology? What about a linear topology (where the last peer is _not_ connected to the first one)? What about a fully connected by random topology? All of these are very important to test.